### PR TITLE
CU-225wvg7 - Update market funding

### PR DESCRIPTION
**Pull request by 0xangelo**
## Issue
[CU-225wvg7](https://app.clickup.com/t/225wvg7)

## Description
Also includes:
- Charging of fees in `open_position`
- Unrealized funding settlement in `open_position`
- Refactoring
- Documentation
- Improvements to the testing infrastructure
- Additional tests for `open_position`

### CODE REVIEW

File: frame/clearing-house/src/lib.rs [hunk 1]
```diff
@@ -592,9 +523,14 @@ pub mod pallet {
 		///
 		/// ## Assumptions or Requirements
 		///
-		/// There's a maximum number of positions ([`Config::MaxPositions`]) than can be open for
-		/// each account id at any given time. If opening a position in a new market exceeds this
-		/// number, the transactions fails.
+		/// - The market must exist and have been initialized prior to calling this extrinsic
+		/// - There's a maximum number of positions ([`Config::MaxPositions`]) than can be open for
```
Comment (Cardosaum) [inline frame/clearing-house/src/lib.rs:527]: "... number of positions _that_ can be ..."

`than` -> `that`

File: frame/clearing-house/src/tests/add_margin.rs [hunk 2]
```diff
@@ -47,14 +50,24 @@ fn deposit_supported_collateral_succeeds() {
 			run_to_block(1);
 			let account = ALICE;
 			let asset = USDC;
-			let amount: Balance = 1_000_u32.into();
+			let amount: Balance = 1_000;
 
-			let before = AccountsMargin::<Runtime>::get(&account).unwrap_or_default();
+			let before = (
+				AccountsMargin::<Runtime>::get(&account).unwrap_or_default(),
+				<AssetsPallet as Inspect<AccountId>>::balance(USDC, &ALICE),
```
Comment (kevin-composable) [inline frame/clearing-house/src/tests/add_margin.rs:57]: maybe just change 'ALICE' to 'account'

Comment (kevin-composable) [reply frame/clearing-house/src/tests/add_margin.rs:57]: same with line 63

File: frame/clearing-house/src/lib.rs [hunk 3]
```diff
@@ -812,30 +798,84 @@ pub mod pallet {
...
+
+			// Pay funding
+			// net position sign | funding rate sign | transfer
+			// --------------------------------------------------------------
+			//                -1 |                -1 | Collateral -> IF
+			//                -1 |                 1 | Fee Pool -> Collateral
+			//                 1 |                -1 | Fee Pool -> Collateral
+			//                 1 |                 1 | Collateral -> IF
+			//                 - |                 0 | n/a
+			//                 0 |                 - | n/a
```
Comment (0xangelo) [inline frame/clearing-house/src/lib.rs:851]: This is wrong actually. All transfers should be between the Collateral and Fee Pool. Maybe I can change this in a future PR?

File: frame/clearing-house/src/tests/add_margin.rs (hunk 2 above)
Comment (0xangelo) [reply frame/clearing-house/src/tests/add_margin.rs:57]: I think I'll introduce `BOB` pretty soon when I start testing the flow of funding payments, so perhaps it's good to maintain a distinctive name

File: frame/clearing-house/src/lib.rs [hunk 4]
```diff
@@ -652,18 +638,13 @@ pub mod pallet {
...
 
 			// Assuming stablecoin collateral and all markets quoted in dollars
-			T::Assets::transfer(
-				asset_id,
-				account_id,
-				&T::PalletId::get().into_account(),
-				amount,
-				true,
-			)?;
+			let pallet_acc = T::PalletId::get().into_sub_account("Collateral");
```
Comment (kevin-composable) [inline frame/clearing-house/src/lib.rs:646]: I would move this into its own function called get_collateral_account or something similar

**Review by kevin-composable — APPROVED** [review]
Really great work. Everything looks good across the board - I had a few small comments but that is it. I would also maybe change where you have abbreviated variable/function names since (1) sometimes it is shortened due to domain knowledge and might not be familiar for others and (2) we have a wide range of english speakers so it might make it easier for all of them to understand if it is more verbose

**Pull request merged**